unit test flaky due to fs event -

I get these on my PR, that do not seem to be related to code i pushed #4202  :
```bash
OK     fsEventsBasic (2.00ms)
##[error]thread '<unnamed>' panicked at 'fs events error: SendError(Ok(FsEvent { kind: "modify", paths: ["/tmp/8259b2ca/file2.txt"] }))', src/libcore/result.rs:1188:5
```

--- a/cli/ops/fs_events.rs
+++ b/cli/ops/fs_events.rs
@@ -22,9 +22,17 @@ pub fn init(i: &mut Isolate, s: &State) {
 }
 
 struct FsEventsResource {
-  #[allow(unused)]
+  paths: Vec<String>,
   watcher: RecommendedWatcher,
   receiver: mpsc::Receiver<Result<FsEvent, ErrBox>>,
+}
+
+impl Drop for FsEventsResource {
+  fn drop(&mut self) {
+    for path in &self.paths {
+      let _ = self.watcher.unwatch(path);
+    }
+  }
 }
 
 /// Represents a file system event.
@@ -76,7 +84,9 @@ pub fn op_fs_events_open(
     Watcher::new_immediate(move |res: Result<NotifyEvent, NotifyError>| {
       let res2 = res.map(FsEvent::from).map_err(ErrBox::from);
       let mut sender = sender.lock().unwrap();
-      futures::executor::block_on(sender.send(res2)).expect("fs events error");
+      // Ignore result, if send failed it means that watcher was already closed,
+      // but not all messages have been flushed.
+      let _ = futures::executor::block_on(sender.send(res2));
     })
     .map_err(ErrBox::from)?;
   let recursive_mode = if args.recursive {
@@ -88,7 +98,11 @@ pub fn op_fs_events_open(
     state.check_read(&PathBuf::from(path))?;
     watcher.watch(path, recursive_mode).map_err(ErrBox::from)?;
   }
-  let resource = FsEventsResource { watcher, receiver };
+  let resource = FsEventsResource {
+    paths: args.paths,
+    watcher,
+    receiver,
+  };
   let table = &mut state.borrow_mut().resource_table;
   let rid = table.add("fsEvents", Box::new(resource));
   Ok(JsonOp::Sync(json!(rid)))

--- a/cli/ops/fs_events.rs
+++ b/cli/ops/fs_events.rs
@@ -22,17 +22,9 @@ pub fn init(i: &mut Isolate, s: &State) {
 }
 
 struct FsEventsResource {
-  paths: Vec<String>,
+  #[allow(unused)]
   watcher: RecommendedWatcher,
   receiver: mpsc::Receiver<Result<FsEvent, ErrBox>>,
-}
-
-impl Drop for FsEventsResource {
-  fn drop(&mut self) {
-    for path in &self.paths {
-      let _ = self.watcher.unwatch(path);
-    }
-  }
 }
 
 /// Represents a file system event.
@@ -98,11 +90,7 @@ pub fn op_fs_events_open(
     state.check_read(&PathBuf::from(path))?;
     watcher.watch(path, recursive_mode).map_err(ErrBox::from)?;
   }
-  let resource = FsEventsResource {
-    paths: args.paths,
-    watcher,
-    receiver,
-  };
+  let resource = FsEventsResource { watcher, receiver };
   let table = &mut state.borrow_mut().resource_table;
   let rid = table.add("fsEvents", Box::new(resource));
   Ok(JsonOp::Sync(json!(rid)))

